src/wp-admin/includes/post.php

```diff
@@ -816,6 +818,11 @@ function post_exists( $title, $content = '', $date = '', $type = '', $status = '
 		$query .= ' AND post_status = %s';
 		$args[] = $post_status;
 	}
+	
+	if ( ! empty( $author ) ) {
+		$query .= ' AND post_author = %s';
```

Comment:
```suggestion
		$query .= ' AND post_author = %d';
```

We can use the `%d` placeholder since we are using an integer value.